frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -511,10 +511,21 @@ export const GRAPH_AXIS_SETTINGS = {
     },
     default: true,
   },
-  "graph.y_axis.auto_range_include_zero": {
-    hidden: true,
+  "graph.y_axis.unpin_from_zero": {
+    section: t`Axes`,
+    group: t`Y-axis`,
+    title: t`Unpin from zero`,
+    widget: "toggle",
+    index: 5,
+    inline: true,
+    getHidden: (series, vizSettings) => {
+      return (
+        !vizSettings["graph.y_axis.auto_range"] ||
+        series[0].card.display === "waterfall"
```

Comment:
Would it also make sense to hide this setting for bar and area charts? Since those by definition will always touch the 0 point on the y-axis?

Reply:
They still can be rendered like this and I've seen charts like this in real life:
<img width="1316" alt="Screenshot 2024-04-29 at 6 02 56 PM" src="https://github.com/metabase/metabase/assets/14301985/5bdfb0b6-01e3-4def-8f3d-95a0abaad2df">

I disliked all of them though